BotApi/BotApi/Services/GraphicService.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CurrencyBot
+{
+    class GraphicService : IGraphicService
+    {
+        public void GetGraphic(List<CurrencyRateDTO> rates)
+        {
+            Bitmap bitmap = new Bitmap(Convert.ToInt32(2160), Convert.ToInt32(2048), PixelFormat.Format32bppArgb);
+            Graphics graph = Graphics.FromImage(bitmap);
+
+            DrawAxes(graph);
+
+            //це вся відстань осі Х
+            TimeSpan MainSpan = (rates[rates.Count - 1].UpdatedAt - rates[0].UpdatedAt);
```

Comment:
Unhandled out of range exception here, if list is empty. You should rather generate empty chart or throw business logic exception.